EIPS/eip-8005.md

```diff
@@ -0,0 +1,144 @@
+---
+eip: 8005
+title: Encrypted Transaction Envelope (ETE)
+description: Typed transaction with encrypted payload and ZK proof for mempool privacy without changing L1 execution.
+author: Tyler Lengyel (@lengyeltyler) <lengyeltyler@proton.me>
+discussions-to: https://ethereum-magicians.org/t/eip-8005-encrypted-transaction-envelope-ete-typed-transaction/25110
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-08-13
+requires: 1559, 2718
+---
+
+## Abstract
+
+This EIP defines a new EIP-2718 typed transaction that encapsulates an Encrypted Transaction Envelope (ETE): a public header, an AEAD-encrypted body, and a zero-knowledge validity proof that binds them. Nodes gossip the opaque, mempool-private payload without exposing calldata. Builders or proposers who opt in decrypt and verify before inclusion, then transform the ETE into a standard EIP-1559 transaction for execution. Consensus rules and EVM semantics remain unchanged. This provides pre-inclusion confidentiality and a standard envelope format that can also be adopted by L2s for end-to-end private execution.
+
+## Motivation
+
+Ethereum provides authenticity and integrity for transactions, but no confidentiality — calldata is broadcast in plaintext, enabling:
+
+- MEV extraction through front-running and sandwich attacks
+- Leakage of trading strategies and private business logic
+- Targeting or harassment of addresses
+
+Private mempools exist, but they are proprietary and incompatible. Standardizing an encrypted envelope:
+
+- Gives wallets and dApps a single, interoperable privacy format
+- Enables zero-knowledge policy proofs (e.g., spending limits, one-per-person) without revealing sensitive details
+- Allows immediate adoption by L2s for full privacy, while keeping L1 backwards-compatible
+
+## Specification
+
+### Transaction Type
+
+This EIP requests assignment of a new EIP-2718 typed transaction (type value **TBD**, to be allocated by editors).
```

Comment:
add a <-- TODO --> html marker for bots to remind you to fill this TBD when moving to review stage in a PR